chaos_genius/alerts/anomaly_alerts.py

```diff
@@ -436,51 +604,90 @@ def _find_point(
         return intended_point
 
     def _format_anomaly_data(
-        self, anomaly_data: List[AnomalyPointOriginal]
+        self, anomaly_data: List[AnomalyPointOriginal], date: datetime.date
     ) -> List[AnomalyPoint]:
         kpi = self._get_kpi()
 
         time_series_freq: Optional[str] = kpi.anomaly_params.get("frequency")
 
+        if time_series_freq == "D":
+            formatted_anomaly_data = self._format_anomaly_data_daily(anomaly_data, date)
+        elif time_series_freq == "H":
+            formatted_anomaly_data = self._format_anomaly_data_hourly(
+                anomaly_data, date
+            )
+        else:
+            raise AlertException(
+                (f"Time series frequency not found or invalid: {time_series_freq}"),
+                alert_id=self.alert_id,
+                kpi_id=self.kpi_id,
+            )
+
+        # Sort in descending order according to severity
+        formatted_anomaly_data.sort(key=lambda point: point.severity, reverse=True)
+
+        return formatted_anomaly_data
+
+    def _format_anomaly_data_daily(
+        self, anomaly_data: List[AnomalyPointOriginal], date: datetime.date
+    ) -> List[AnomalyPoint]:
+        formatted_anomaly_data: List[AnomalyPoint] = []
+
         # get previous anomaly point for comparison
-        time_diff = datetime.timedelta(days=1, hours=0, minutes=0)
         prev_day_data = self._get_anomalies(
-            time_diff=time_diff, anomalies_only=False, include_severity_cutoff=False
+            this_date_only=date - datetime.timedelta(days=1, hours=0, minutes=0),
+            anomalies_only=False,
+            include_severity_cutoff=False,
+        )
+
+        formatted_anomaly_data: List[AnomalyPoint] = []
+        for point in anomaly_data:
+            # daily granularity - find point in the previous day
+            previous_point = self._find_point(point, prev_day_data)
+
+            formatted_anomaly_data.append(
+                AnomalyPoint.from_original(point, previous_point)
+            )
+
+        return formatted_anomaly_data
+
+    def _format_anomaly_data_hourly(
+        self, anomaly_data: List[AnomalyPointOriginal], date: datetime.date
+    ) -> List[AnomalyPoint]:
+        formatted_anomaly_data: List[AnomalyPoint] = []
+
+        # get current day's data for comparison
+        cur_day_data = self._get_anomalies(
+            this_date_only=date, anomalies_only=False, include_severity_cutoff=False
+        )
+        # get previous day data too.
+        # used only when a point is present for 00:00
+        #  - the previous point in that case will be previous day's 11PM data
+        prev_day_data = self._get_anomalies(
+            this_date_only=date - datetime.timedelta(days=1, hours=0, minutes=0),
+            anomalies_only=False,
+            include_severity_cutoff=False,
```

Comment:
Some optimization could be done to get data >= 11PM of previous day, instead of whole of previous day